python/vineyard/__init__.py

```diff
@@ -398,3 +398,54 @@ def connect(*args, **kwargs):
         )
         try_init()
     return _connect(*args, **kwargs)
+
+
+def write(
+    value: Any,
+    path: str,
+):
+    """
+    Write python value to vineyard.
+    Notice, the API is only used for the integration with kubernetes
+    volume.
+
+    Parameters:
+        client: IPCClient
+            The vineyard client to use.
+        path: str
+            The path that represents a vineyard object.
+
+    .. code:: python
+
+        >>> arr = np.arange(8)
+        >>> client.write(arr)
+    """
+    socket_path = os.path.join(path, "vineyard.sock")
+    client = connect(socket_path)
+    client.put(value, persist=True, name=path)
+
+
+def read(
+    path: str,
+):
+    """
+    Read vineyard object from path, and return python value.
+    Notice, the API is only used for the integration with kubernetes
+    volume.
+
+    Parameters:
+        path: str
+            The path that represents a vineyard object.
+
+    .. code:: python
+
+        >>> arr = vineyard.read('/a/b/c/d/f')
+        >>> arr
+        array([0, 1, 2, 3, 4, 5, 6, 7])
+
+    Returns:
+        A python object that return by the resolver, by resolving an vineyard object.
+    """
+    socket_path = os.path.join(path, "vineyard.sock")
+    client = connect(socket_path)
+    return client.get(name=path)
```

Comment:
Add a module `vineyard/csi/__init__.py` and put those two functions here, as they do strong assumption about the value of directory (where to find the socket and what should be used as names).

And, do check if the socket exists, and throw errors like "seems that the given path  is not generated by the vineyard CSI driver", or something like that.

Reply:
Get it.